test/src/e2e_vm_tests/test_programs/should_pass/stdlib/result/src/main.sw

```diff
@@ -7,6 +7,7 @@ fn main() -> bool {
     test_ok();
     test_err();
     test_unwrap_ok();
+    test_unwrap_or();
```

Comment:
Do you mind expanding this whole test for `Result` into a more elaborate one similarly to what we did for `Option` here https://github.com/FuelLabs/sway/pull/3322?